### PR TITLE
Support level 1 headings in markdown

### PR DESCRIPTION
**Pull request by gaborcsardi**
They generate separate sections.

This PR is on top of #907.

I'll add docs later, once we make sure that this is what we want.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Rebased.... it addition to the docs, I'll also add better tests for edge cases, like #909.

File: R/rd.R [hunk 1]
```diff
@@ -41,16 +41,16 @@ roclet_tags.roclet_rd <- function(x) {
...
+    details = tag_markdown_with_sections,
     docType = tag_name,
     encoding = tag_value,
     evalRd = tag_code,
     example = tag_value,
     examples = tag_examples,
     family = tag_value,
     field = tag_name_description,
-    format = tag_markdown,
+    format = tag_markdown_with_sections,
```
Comment (hadley) [inline R/rd.R]: I think we should only allow in `@description` and `@details` for simplicity

File: R/rd.R [hunk 2]
```diff
@@ -243,7 +243,10 @@ topic_add_simple_tags <- function(topic, block) {
   tag_names <- names(block)[is_simple]
 
   for (i in seq_along(tag_values)) {
-    topic$add_simple_field(tag_names[[i]], tag_values[[i]])
+    topic$add_simple_field(tag_names[[i]], tag_values[[i]][1])
```
Comment (hadley) [inline R/rd.R]:
```suggestion
    topic$add_simple_field(tag_names[[i]], tag_values[[i]][[1]])
```

File: R/tag.R [hunk 3]
```diff
@@ -241,3 +241,21 @@ tag_markdown <- function(x) {
   x$val <- markdown_if_active(x$val, x)
   tag_value(x)
 }
+
+#' @export
+#' @rdname roxy_tag
+tag_markdown_with_sections <- function(x) {
+  x$val <- markdown_if_active(x$val, x, sections = TRUE)
+  if (length(x$val) == 1 && x$val[[1]] == "") {
+    roxy_tag_warning(x, "requires a value")
```
Comment (hadley) [inline R/tag.R]: Can you use the same test as above? And it needs to return, so that further processing doesn't occur.

File: R/tag.R [hunk 4]
```diff
@@ -241,3 +241,21 @@ tag_markdown <- function(x) {
...
+  }
+  for (i in seq_along(x$val)) {
+    if (!rdComplete(x$val[i])) {
```
Comment (hadley) [inline R/tag.R]: Why doesn't this match `tag_markdown()`?

File: R/rd.R (hunk 1 above)
Comment (gaborcsardi) [reply R/rd.R]: OK.

File: R/rd.R (hunk 2 above)
Comment (gaborcsardi) [reply R/rd.R]: `tag_values[[i]]` is a character vector now, for tags with sections. This code is not the easiest to read, so we can make sections more explicit if you want. Then we'll need a `topic_add_tags_with_sections()` function I guess.

File: R/tag.R (hunk 3 above)
Comment (gaborcsardi) [reply R/tag.R]: Hmmm, which test? As in `tag_value()`? `x$val` can be a vector now, so it needs a slightly different test.

File: R/tag.R (hunk 4 above)
Comment (gaborcsardi) [reply R/tag.R]: Because `x$val` is a vector now, First element is the regular content, the rest are the sections. But again, we can make this more explicit by returning a list.

File: R/rd.R (hunk 1 above)
Comment (gaborcsardi) [reply R/rd.R]: done.

File: R/tag.R (hunk 3 above)
Comment (hadley) [reply R/tag.R]: I think it would be natural to match `tag_examples()`; i.e. check `x$val == ""` before parsing the markdown.

File: R/tag.R (hunk 4 above)
Comment (hadley) [reply R/tag.R]: But `tag_markdown()` doesn't perform this check at all. Why not?

File: R/tag.R (hunk 3 above)
Comment (gaborcsardi) [reply R/tag.R]: oh, ok, that makes sense.

File: R/tag.R (hunk 4 above)
Comment (gaborcsardi) [reply R/tag.R]: It does I think, it calls `tag_value()`, that does these checks.

Comment (hadley) [reply R/tag.R]: Ah got it.

**Comment by hadley** [comment]
I just pushed a little tweak. I'll merge once you update the docs.

**Comment by gaborcsardi** [comment]
OK, added short docs in the Rd formatting vignette.

**Comment by gaborcsardi** [comment]
@hadley also rebased, so it is ready, I think.

**Comment by hadley** [comment]
Thanks!
